controllers/comments.controller.js

```diff
@@ -17,4 +18,27 @@ exports.getCommentsByArticleId = (req, res, next) => {
     .catch((err) => {
         next(err)
     })
-}
\ No newline at end of file
+}
+
+  exports.postCommentById = (req, res, next) => {
+    const { body: commentBody } = req;
+    const { article_id } = req.params;
+  
+    if (!commentBody.username || !commentBody.body) {
+      return res.status(400).send({ msg: 'Bad request' });
+    }
+  
+    checkIfArticleExists(article_id)
+      .then(() => {
+        return checkIfUserExists(commentBody.username);
```

Comment:
The username has to exist in the users table so if it does not PSQL will reject the query so you can handle this error this way - catch the error and pass to psql error handler